pkg/input/cloudapi/resources.go

```diff
@@ -0,0 +1,115 @@
+package cloudapi
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/google/go-querystring/query"
+)
+
+type (
+	ResourcesParameters struct {
+		EnvironmentID []string `url:"environment_id,omitempty"`
+		ResourceType  []string `url:"resource_type,omitempty"`
+		ResourceID    []string `url:"resource_id,omitempty"`
+		NativeID      []string `url:"native_id,omitempty"`
+		ID            []string `url:"id,omitempty"`
+		Platform      []string `url:"platform,omitempty"`
+		Name          []string `url:"name,omitempty"`
+		Location      []string `url:"location,omitempty"`
+	}
+
+	CollectionDocumentRes struct {
+		Data  []ResourceObject `json:"data"`
+		Links Links
+	}
+
+	ResourceObject struct {
+		ID         string             `json:"id,omitempty"`
+		Type       string             `json:"type"`
+		Attributes ResourceAttributes `json:"attributes,omitempty"`
+	}
+
+	ResourceAttributes struct {
+		Namespace    string                 `json:"namespace"`
+		ResourceType string                 `json:"resource_type"`
+		ResourceID   string                 `json:"resource_id"`
+		State        map[string]interface{} `json:"state"`
+		Tags         map[string]interface{} `json:"tags"`
+	}
+
+	Links struct {
+		Next string `json:"next"`
+	}
+)
+
+func (c *Client) Resources(ctx context.Context, orgID string, params ResourcesParameters) (resources []ResourceObject, e error) {
+	url := fmt.Sprintf("%s/rest/orgs/%s/cloud/resources", c.url, orgID)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+	q, err := query.Values(params)
+	if err != nil {
+		return nil, err
+	}
+	q.Add("version", c.version)
+	// We're hardcoding cloud here just to simplify things for now. If we do end
+	// up with a use-case for IaC resources from the Cloud API, we'll need to
+	// keep the input types in mind and produce multiple inputs from the
+	// response.
+	q.Add("kind", "cloud")
+	req.URL.RawQuery = q.Encode()
+
+	results, err := c.resourcesPage(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	resources = append(resources, results.Data...)
+
+	for results.Links.Next != "" {
+		url := fmt.Sprintf("%s/%s", c.url, results.Links.Next)
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+		if err != nil {
+			return nil, err
+		}
+		results, err = c.resourcesPage(ctx, req)
+		if err != nil {
+			return nil, err
+		}
+		resources = append(resources, results.Data...)
+	}
+
+	return resources, nil
+}
+
+func (c *Client) resourcesPage(ctx context.Context, req *http.Request) (results CollectionDocumentRes, e error) {
+	req.Header.Set("Content-Type", "application/vnd.api+json")
+	req.Header.Set("Authorization", c.authorization)
+
+	res, err := c.httpClient.Do(req)
+	if err != nil {
+		return results, err
```

Comment:
Nitpick: I don't like the fact that we're mixing named returns as well as explicit ones; IMO we should rename `e` to `err` and just use a parameter-less plain `return` here and below (and assign to `results`/`err` as appropriate).

Reply:
Thanks! I think that mix centered around the deferred IIFE below, which was intended to catch errors from closing the request body. I think those errors are both rare and unimportant, so I've switched the defer to the more idiomatic one and unnamed the return variables in f75aac7.